typing/mode_intf.mli

```diff
@@ -58,13 +58,25 @@ type equate_step =
   | Left_le_right
   | Right_le_left
 
+(** Simple mode error specific to axis whose carrier type is ['a]. [left] is the lower
+bound of actual mode and [right] is the upper bound of expected mode. [left <= right] is
+false, which is why the submode failed. *)
+type 'a simple_error =
+  { left : 'a;
+    right : 'a
```

Comment:
Better call these `actual` and `expected` to align with the printing in `module Report`.